Restructure secure settings docs and minor additions (#5965)
elastic/cloud-on-k8s: docs/orchestrating-elastic-stack-applications/elasticsearch/es-secure-settings.asciidoc

= Secure settings

You can specify link:https://www.elastic.co/guide/en/elasticsearch/reference/current/secure-settings.html[secure settings] with Kubernetes secrets.

> The secrets should contain a key-value pair for each secure setting you want to add. ECK automatically injects these settings into the keystore on each Elasticsearch node before it starts Elasticsearch. The ECK operator keeps watching the secrets for changes and will update the Elasticsearch keystore when it detects a change.

Reviewer: ```suggestion
The secrets should contain a key-value pair for each secure setting you want to add. ECK automatically injects these settings into the keystore on each Elasticsearch node before it starts Elasticsearch. The ECK operator continues to watch the secrets for changes and will update the Elasticsearch keystore when it detects a change.
```

  gcs_client_2: RWxhc3RpYyBDbG91ZCBvbiBLOHMgKEVDSykgLSBHQ1MgY2xpZW50IDIK
----

> == More Examples

Reviewer 2: ```suggestion
== More examples
```

> Check <<{p}-snapshots,How to create automated snapshots>> for an example use case that illustrates how secure settings can be used to set up automated Elasticsearch snapshots to an GCS storage bucket.

Reviewer 3: ```suggestion
Check <<{p}-snapshots,How to create automated snapshots>> for an example use case that illustrates how secure settings can be used to set up automated Elasticsearch snapshots to a GCS storage bucket.
```
